DataFormats/ParticleFlowReco/interface/PFTrajectoryPoint.h

```diff
@@ -52,10 +52,35 @@ namespace reco {
       HOLayer = 8,
       /// VFcal(HF) front face
       VFcalEntrance = 9,
-
-      NLayers = 10
+      // Number of valid layers
+      NLayers = 10,
+      // Unknown
+      Unknown = -1
     };
 
+    static std::map<std::string, LayerType> create_map() {
+      std::map<std::string, LayerType> m;
+      m["ClosestApproach"] = LayerType::ClosestApproach;
+      m["BeamPipeOrEndVertex"] = LayerType::BeamPipeOrEndVertex;
+      m["PS1"] = LayerType::PS1;
+      m["PS2"] = LayerType::PS2;
+      m["ECALEntrance"] = LayerType::ECALEntrance;
+      m["ECALShowerMax"] = LayerType::ECALShowerMax;
+      m["HCALEntrance"] = LayerType::HCALEntrance;
+      m["HCALExit"] = LayerType::HCALExit;
+      m["HOLayer"] = LayerType::HOLayer;
+      m["VFcalEntrance"] = LayerType::VFcalEntrance;
+      return m;
+    }
+    static LayerType layerTypeFromString(std::string layerTypeString) {
+      std::map<std::string, LayerType> layerTypeMap = create_map();
```

Comment:
I think it might be preferable for a few reasons to define the map as a constant within this function:
```cpp
const std::map<std::string, LayerType> layerTypeMap{
  {"ClosestApproach", LayerType::ClosestApproach},
  {"BeamPipeOrEndVertex", LayerType::BeamPipeOrEndVertex},
  {"PS1", LayerType::PS1},
  {"PS2", LayerType::PS2},
  {"ECALEntrance", LayerType::ECALEntrance},
  {"ECALShowerMax", LayerType::ECALShowerMax},
  {"HCALEntrance", LayerType::HCALEntrance},
  {"HCALExit", LayerType::HCALExit},
  {"HOLayer", LayerType::HOLayer},
  {"VFcalEntrance", LayerType::VFcalEntrance},
};
```

Reply:
This looks like a very expensive way to map a name of an enum to the enum (even with the pre-constructed `std::map`). Linear search over a static array would require less memory and likely be faster (see e.g. 
https://github.com/cms-sw/cmssw/blob/431e2fdfedec052e73c16e9f06de98ade41ebc56/DataFormats/TrackReco/src/TrackBase.cc#L137-L142
for an example)

Reply:
Okay. This will be called only at the beginning of each job 1-3 times, and not for each event, so I don't think it's critical to make it less expensive, but let me think..

Reply:
If only used in job startup the `std::map` could be acceptable (assuming it gets destroyed after last use to not "hoard" memory). This leaves the main argument against to be "bad example for others".

Reply:
Thanks. At the end, I just adopted your string to enum conversion method. I agree that it's nicer.